pkg/pillar/pubsub/pubsub.go

```diff
@@ -788,12 +789,14 @@ func (pub *Publication) GetAll() map[string]interface{} {
 //  foo := s1.Get(key)
 //  fooAll := s1.GetAll()
 
+type SubCreateHandler func(ctx interface{}, key string, status interface{})
```

Comment:
Instead of defining one new identical type, how about defining using a SubHandler for all of create, modify, and delete?